### ISSUE TITLE
CORS CORS CORS

### ISSUE THREAD

**Issue by Zyles** [esi/esi-issues#1163]
# Bug

CORS errors all around.

## Request

OPTIONS https://esi.evetech.net/latest/universe/types/12551?datasource=tranquility 502

## Response

Access to XMLHttpRequest at 'https://esi.evetech.net/latest/universe/types/12551?datasource=tranquility' from origin 'http://localhost:8080' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

### Status Code

`502`

### Headers

Donno, didn't save them.

### Body

None.

## Expected

Working API call.

# Checklist

Check all boxes that apply to this issue:

- [x] Bug description is provided
- [x] Request path is provided
- [x] Response status code is provided
- [ ] Response headers are provided
- [ ] Response body is provided
- [x] Expected response is provided

**Comment by Blacksmoke16** [comment]
Not sure what the problem is?

```bash
curl -I https://esi.evetech.net/latest/universe/types/12551?datasource=tranquility
HTTP/2 200 
date: Mon, 23 Dec 2019 21:38:17 GMT
content-type: application/json; charset=UTF-8
access-control-allow-credentials: true
access-control-allow-headers: Content-Type,Authorization,If-None-Match,X-User-Agent
access-control-allow-methods: GET,HEAD,OPTIONS
access-control-allow-origin: *
access-control-expose-headers: Content-Type,Warning,ETag,X-Pages,X-ESI-Error-Limit-Remain,X-ESI-Error-Limit-Reset
access-control-max-age: 600
allow: GET,HEAD,OPTIONS
cache-control: public
content-language: en-us
etag: W/"bd65f9cfd29573bc513c5d5ef2260ca532f7ef1a4944cfa635e18e0b"
expires: Tue, 24 Dec 2019 11:05:00 GMT
last-modified: Mon, 23 Dec 2019 11:02:49 GMT
strict-transport-security: max-age=31536000
vary: Accept-Language
x-esi-error-limit-remain: 100
x-esi-error-limit-reset: 43
x-esi-request-id: fc7d62a4-56f5-45e3-9f72-f63a95024b2b
```

Seems to be just fine?  Going to close unless you have a reproducible example.

**Comment by Blacksmoke16** [comment]
This is most likely a duplicate of https://github.com/esi/esi-issues/issues/1082 due to the `502` error you got.

**Comment by Zyles** [comment]
The problem is it has been one year and it has not been resolved yet.

Put up some external tests vs the API and you could probably find a lot of problems. But if you never use it, it is flawless.

**Comment by guiguilechat** [comment]
Hmm I think you are the only one having this issue.
Also you don't give any help on what is the issue. Just claim there is an issue ; people can't reproduce it, but you refuse to give more explanation. 

Why would people waste time on an error that may or may not be from our side, that they can't reproduce, that basically they can't fix because YOU don't want to take the time to make a proper explanation ?

Sorry but "CORS errors all around." is not a correct bug explanation. I just tried your URL and have no CORS error.I look closely at the errors I have when accessing ESI and I can't remember a time I had such an error.

**Comment by Blacksmoke16** [comment]
> Hmm I think you are the only one having this issue.

@guiguilechat It's an actual problem.  This issue however is a duplicate of #1082, hence why I closed it.

**Comment by Zyles** [comment]
> Hmm I think you are the only one having this issue.
> Also you don't give any help on what is the issue. Just claim there is an issue ; people can't reproduce it, but you refuse to give more explanation.
> 
> Why would people waste time on an error that may or may not be from our side, that they can't reproduce, that basically they can't fix because YOU don't want to take the time to make a proper explanation ?
> 
> Sorry but "CORS errors all around." is not a correct bug explanation. I just tried your URL and have no CORS error.I look closely at the errors I have when accessing ESI and I can't remember a time I had such an error.

If you in your right mind think that typing in the URL in your browser will create a CORS error. Then you obviously don't understand how CORS works and what it is.

That is why you have a hard time understanding what the problem is and it does not require more explanation to someone who has a clue.

I suggest using Google to read up on it as it does not have a place in a github issue.
